test/y2storage/blk_device_test.rb

```diff
@@ -345,4 +345,86 @@
       end
     end
   end
+
+  describe "#udev_full_label" do
+    let(:device_name) { "/dev/sda1" }
+    before { allow(device).to receive(:filesystem_label).and_return(label) }
+
+    context "for devices without label" do
```

Comment:
The tests still check only for `filesystem_label.nil?` which I'm not sure covers all the possibilities. What about some tests less coupled with the implementation? I mean, really using a device:

* Without filesystem
* With a filesystem that has no label (an empty label, nil is not possible here)
* With a labeled filesystem

Instead of just mocking the call to `filesystem_label`, which is an implementation detail. The very same methods could be implemented not relying on `filesystem_label`. I prefer mocking the whole situation when possible instead of coupling the current implementation and the tests so much.